### PR TITLE
allow CMD on images to be overriden

### PR DESCRIPTION
**Pull request by fabiokung**
### Description of the Change

The control plane will set the new `Process` field for jobs that specify a custom command. The new (array) entrypoint and command follow the same rules as Docker.

The old flat string entrypoint has precedence, to allow for a smooth transition without breaking existing jobs. The rollout plan described at Netflix/titus-api-definitions#52

### CODE REVIEW

File: executor/runtime/types/types_test.go [hunk 1]
```diff
@@ -23,10 +27,71 @@ func TestEntrypointParse(t *testing.T) {
...
+			TitusInfo: &titus.ContainerInfo{
+				Process: &titus.ContainerInfo_Process{
+					Entrypoint: entrypoint,
+					Command:    []string{"sleep", "1"},
+				},
+			},
+		}
+
+		entrypoint, cmd, err := c.Process()
+		if err != nil {
```
Comment (sargun) [inline executor/runtime/types/types_test.go]: `require.NoError(err)`?

File: executor/runtime/types/types_test.go [hunk 2]
```diff
@@ -23,10 +27,71 @@ func TestEntrypointParse(t *testing.T) {
...
+	assert.Len(t, cmd, 0)
+}
+
+func TestDefaultIsNil(t *testing.T) {
+	c := Container{
+		TitusInfo: &titus.ContainerInfo{},
+	}
+
+	entrypoint, cmd, err := c.Process()
+	if err != nil {
```
Comment (sargun) [inline executor/runtime/types/types_test.go]: `require.NoError`

File: hack/test-images/shell-entrypoint/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,4 @@
+FROM ubuntu:xenial
```
Comment (sargun) [inline hack/test-images/shell-entrypoint/Dockerfile]: Please add instructions to build this in the .circle config.

File: .gitignore [hunk 4]
```diff
@@ -94,5 +94,5 @@ checkstyle-result.xml
 # unignore directories named titus-executor
 !*/titus-executor/
 
-test-standalone.log
```
Comment (sargun) [inline .gitignore]: Why did you remove these two?

File: hack/test-images/shell-entrypoint/Dockerfile (hunk 3 above)
Comment (sargun) [reply hack/test-images/shell-entrypoint/Dockerfile]: Err, call the build command

File: .gitignore (hunk 4 above)
Comment (fabiokung) [reply .gitignore]: the IDE told me to, since there's another rule for `*.log`

File: executor/runtime/types/types_test.go (hunk 1 above)
Comment (fabiokung) [reply executor/runtime/types/types_test.go]: no other tests here are using that lib. I'd personally rather not to, unless you feel strongly about it.

File: .gitignore (hunk 4 above)
Comment (sargun) [reply .gitignore]: neat.
